src/uct/ib/base/ib_md.h

```diff
@@ -30,6 +30,28 @@
 #define UCT_IB_CONFIG_PREFIX      "IB_"
 
 
+/**
+ * IB transport registration routines
+ *
+ * @param _component      Component to add the transport to
+ * @param _name           Name of the transport (should be a token, not a string)
+ * @param _query_devices  Function to query the list of available devices
+ * @param _iface_class    Struct type defining the uct_iface class
+ * @param _cfg_prefix     Prefix for configuration variables
+ * @param _cfg_table      Transport configuration table
+ * @param _cfg_struct     Struct type defining transport configuration
+ */
+#define UCT_IB_TL_DEFINE(_component, _name, _query_devices, _iface_class, \
+                         _cfg_prefix, _cfg_table, _cfg_struct) \
+    UCT_TL_REGISTER_DEF(_component, _name, _query_devices, _iface_class, \
+                        _cfg_prefix, _cfg_table, _cfg_struct) \
+    void uct_tl_##_name##_init(void) { \
```

Comment:
should we use UCT_TL_INIT macro?

Reply:
> should we use UCT_TL_INIT macro?

no: UCT_TL_INIT macro initialises both component and TL, but in IB component is initialised separately (single component for multiple TL's)

Reply:
so maybe UCT_TL_INIT should be something like UCT_SINGLE_TL_INIT (need to think of a good name) since it does not work for every TL.. 
can we share some code/macro between IB TL init and other TLs init?

Reply:
What about Cuda/RoCM/uGNI - are they also single components?

Reply:
uGNI has multiple TL's per component
UCT_TL_INIT_ONLY - what about this name?

Reply:
> so maybe UCT_TL_INIT should be something like UCT_SINGLE_TL_INIT (need to think of a good name) since it does not work for every TL..
> can we share some code/macro between IB TL init and other TLs init?

renamed to UCT_SINGLE_TL_INIT